binary-option/program/src/processor.rs

```diff
@@ -0,0 +1,743 @@
+
+    let collector_long_token_account: Account =
+        assert_initialized(collector_long_token_account_info)?;
+    let collector_short_token_account: Account =
+        assert_initialized(collector_short_token_account_info)?;
+    let collector_account: Account = assert_initialized(collector_account_info)?;
+    let escrow_account: Account = assert_initialized(escrow_account_info)?;
+    let mut binary_option = BinaryOption::try_from_slice(&pool_account_info.data.borrow_mut())?;
+
+    // Get program derived address for escrow
+    let (escrow_owner_key, bump_seed) = Pubkey::find_program_address(
+        &[
+            long_token_mint_info.key.as_ref(),
+            short_token_mint_info.key.as_ref(),
+            token_program_info.key.as_ref(),
+            program_id.as_ref(),
+        ],
+        program_id,
+    );
+    let seeds = &[
+        long_token_mint_info.key.as_ref(),
+        short_token_mint_info.key.as_ref(),
+        token_program_info.key.as_ref(),
+        program_id.as_ref(),
+        &[bump_seed],
+    ];
+
+    if !binary_option.settled {
+        return Err(BinaryOptionError::BetNotSettled.into());
+    }
+    assert_owned_by(long_token_mint_info, &spl_token::id())?;
+    assert_owned_by(short_token_mint_info, &spl_token::id())?;
+    assert_keys_equal(collector_long_token_account.owner, *collector_info.key)?;
+    assert_keys_equal(collector_short_token_account.owner, *collector_info.key)?;
+    assert_keys_equal(collector_account.owner, *collector_info.key)?;
+    assert_keys_equal(escrow_owner_key, *escrow_authority_info.key)?;
+    assert_keys_equal(
+        *long_token_mint_info.key,
+        binary_option.long_mint_account_pubkey,
+    )?;
+    assert_keys_equal(
+        *short_token_mint_info.key,
+        binary_option.short_mint_account_pubkey,
+    )?;
+    assert_keys_equal(*escrow_account_info.key, binary_option.escrow_account_pubkey)?;
+    assert_keys_equal(*escrow_account_info.key, binary_option.escrow_account_pubkey)?;
+    assert_keys_equal(
+        collector_long_token_account.mint,
+        binary_option.long_mint_account_pubkey,
+    )?;
+    assert_keys_equal(
+        collector_short_token_account.mint,
+        binary_option.short_mint_account_pubkey,
+    )?;
+    assert_keys_equal(
+        collector_account.mint,
+        binary_option.escrow_mint_account_pubkey,
+    )?;
+
+    let winner_long = collector_long_token_account.mint == binary_option.winning_side_pubkey;
+    let winner_short = collector_short_token_account.mint == binary_option.winning_side_pubkey;
+    let reward = match [winner_long, winner_short] {
+        [true, false] => collector_long_token_account.amount,
+        [false, true] => collector_short_token_account.amount,
+        _ => return Err(BinaryOptionError::TokenNotFoundInPool.into()),
+    };
+
+    topup(
+        escrow_authority_info,
+        rent_info,
+        system_account_info,
+        fee_payer_info,
+        1,
+    )?;
```

Comment:
allows an arbitrary account to crank the collect instruction by sending funds to the escrow account (paying for rent).

Reply:
This shouldn't be needed -- the escrow authority info is the PDA, which is just used for signatures.  It doesn't need rent to stay around, and shouldn't be given funds.

Reply:
python client was breaking without this

Reply:
I think the reason is that you need to be able to pay to invoke `Burn`

Reply:
What was the error being thrown? The transaction is paid for at the start, no need to pay from the PDA

Reply:
Insufficient funds pops up when invoking burn. This is because the escrow authority pays for that instruction (I think). I might be totally wrong here, but I remember everything breaking when I removed this. Can try again to confirm.

Reply:
Pleasantly surprised to be wrong! No idea why this was breaking before, but I'll remove this